esmvalcore/preprocessor/_derive/_shared.py

```diff
@@ -170,15 +170,15 @@ def _create_pressure_array(cube, ps_cube, top_limit):
 def _get_pressure_level_widths(array, air_pressure_axis=1):
     """Compute pressure level widths.
 
-    For a 1D array with pressure level columns, return a 1D array with
+    For a 4D array with pressure level columns, return a 4D array with
```

Comment:
Is 4D mandatory or does it work for other dimensions (given that air_pressure is available)?

Reply:
You're right, this should work for any array